src/com/carolinarollergirls/scoreboard/defaults/DefaultScoreBoardModel.java

```diff
@@ -185,144 +179,234 @@ public void startJam() {
 		synchronized (runLock) {
 			if (!getClock(Clock.ID_JAM).isRunning()) {
 				createSnapshot(ACTION_START_JAM);
-				
-				requestBatchStart();
-				ClockModel pc = getClockModel(Clock.ID_PERIOD);
-				ClockModel jc = getClockModel(Clock.ID_JAM);
-				ClockModel tc = getClockModel(Clock.ID_TIMEOUT);
-				ClockModel ic = getClockModel(Clock.ID_INTERMISSION);
-
-				ic.stop();
-				pc.start();
-
-				if (!jc.isTimeAtStart())
-					jc.changeNumber(1);
-				jc.resetTime();
-				jc.start();
-
-				tc.stop();
-
-				getTeamModel("1").startJam();
-				getTeamModel("2").startJam();
-				requestBatchEnd();
-
+				_startJam();
 				ScoreBoardManager.gameSnapshot();
 			}
 		}
 	}
-	public void stopJam() {
+	public void stopJamTO() {
 		synchronized (runLock) {
-			if (getClockModel(Clock.ID_JAM).isRunning()) {
-				_stopJam();
-			} else if (getClockModel(Clock.ID_TIMEOUT).isRunning()) {
-				_stopTimeout();
-			} else if (!getClockModel(Clock.ID_LINEUP).isRunning()) {
+			ClockModel jc = getClockModel(Clock.ID_JAM);
+			ClockModel lc = getClockModel(Clock.ID_LINEUP);
+			ClockModel tc = getClockModel(Clock.ID_TIMEOUT);
+
+			if (jc.isRunning()) {
+				ScoreBoardManager.gameSnapshot(true);
+				createSnapshot(ACTION_STOP_JAM);
+				_endJam(false);
+			} else if (tc.isRunning()) {
+				createSnapshot(ACTION_STOP_TO);
+				_endTimeout();
+			} else if (!lc.isRunning()) {
+				createSnapshot(ACTION_LINEUP);
 				_startLineup();
 			}
 		}
 	}
-	private void _stopJam() {
+	public void timeout() { 
 		synchronized (runLock) {
-			ScoreBoardManager.gameSnapshot(true);
-			createSnapshot(ACTION_STOP_JAM);
-
-			ClockModel pc = getClockModel(Clock.ID_PERIOD);
-			ClockModel jc = getClockModel(Clock.ID_JAM);
-			ClockModel lc = getClockModel(Clock.ID_LINEUP);
+			createSnapshot(ACTION_TIMEOUT);
 
 			requestBatchStart();
```

Comment:
You use a batch here for a single call that also starts a batch, but not in stopJamTO which does similar.